File: src/commonMain/kotlin/com/xebia/functional/loaders/BaseLoader.kt
package com.xebia.functional.loaders

import com.xebia.functional.domain.Document
import com.xebia.functional.textsplitters.BaseTextSplitter

interface BaseLoader {
    suspend fun load(): List<Document>
    suspend fun loadAndSplit(textSplitter: BaseTextSplitter): List<Document>
}

File: scala/src/main/scala/com/xebia/functional/loaders/BaseLoader.scala
package com.xebia.functional.scala.loaders

import cats.effect.Resource

import com.xebia.functional.scala.domain.Document
import com.xebia.functional.scala.textsplitters.BaseTextSplitter

trait BaseLoader[F[_]]:
  def load: F[List[Document]]
  def loadAndSplit(textSplitter: BaseTextSplitter[F]): F[List[Document]]
